module-apps/application-messages/windows/OptionsMessages.hpp

```diff
@@ -5,10 +5,35 @@
 
 #include "ApplicationMessages.hpp"
 
+#include <OptionWindow.hpp>
 #include <Interface/SMSRecord.hpp>
 #include <Option.hpp>
 
-std::list<gui::Option> smsWindowOptions(app::ApplicationMessages *app, const SMSRecord &record);
+namespace gui
+{
+
+    class SmsWindowOptions : public OptionWindow
+    {
+      public:
+        explicit SmsWindowOptions(app::ApplicationCommon *app, std::string windowName);
+        void onBeforeShow(ShowMode mode, SwitchData *data) override;
+
+      private:
+        enum class ValidCodes
```

Comment:
maybe move it to Option Style -> it is shared between windows. We have there:
```
    enum class ContactOperation
    {
        Add,
        Details,
        Edit
    };
```
Maybe name this enum OptionRecordValidity ?

Reply:
done